internal/services/hybridcompute/arc_machine_extension_resource_test.go

```diff
@@ -0,0 +1,183 @@
+	resp, err := client.Get(ctx, *id)
+	exists := false
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return &exists, nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r HybridComputeMachineExtensionResource) basic(data acceptance.TestData, template string) string {
+	return fmt.Sprintf(`
+				%s
+
+resource "azurerm_arc_machine_extension" "test" {
+  name                      = "acctest-hcme-%d"
+  arc_machine_id            = data.azurerm_arc_machine.test.id
+  publisher                 = "Microsoft.Azure.Monitor"
+  type                      = "AzureMonitorLinuxAgent"
+  location                  = "%s"
+}
+`, template, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r HybridComputeMachineExtensionResource) requiresImport(basicConfig string) string {
+	return fmt.Sprintf(`
+			%s
+
+resource "azurerm_arc_machine_extension" "import" {
+  name                      = azurerm_arc_machine_extension.test.name
+  arc_machine_id            = azurerm_arc_machine_extension.test.arc_machine_id
+  publisher                 = azurerm_arc_machine_extension.test.publisher
+  type                      = azurerm_arc_machine_extension.test.type
+  location                  = azurerm_arc_machine_extension.test.location
+}
+`, basicConfig)
+}
+
+func (r HybridComputeMachineExtensionResource) complete(data acceptance.TestData, template string) string {
+	return fmt.Sprintf(`
+			%s
+
+resource "azurerm_arc_machine_extension" "test" {
+  name                               = "acctest-hcme-%d"
+  arc_machine_id                     = data.azurerm_arc_machine.test.id
+  location                           = "%s"
+  automatic_upgrade_enabled          = false
+  publisher                          = "Microsoft.Azure.Monitor"
+  type                               = "AzureMonitorLinuxAgent"
+  type_handler_version               = "1.24"
+}
+`, template, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r HybridComputeMachineExtensionResource) update(data acceptance.TestData, template string) string {
+	return fmt.Sprintf(`
+			%s
+
+resource "azurerm_arc_machine_extension" "test" {
+  name                               = "acctest-hcme-%d"
+  arc_machine_id                     = data.azurerm_arc_machine.test.id
+  location                           = "%s"
+  automatic_upgrade_enabled          = true
+  publisher                          = "Microsoft.Azure.Monitor"
+  type                               = "AzureMonitorLinuxAgent"
+  type_handler_version               = "1.25"
+}
+`, template, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r HybridComputeMachineExtensionResource) template(data acceptance.TestData) string {
+	d := HybridComputeMachineDataSource{}
+	clientSecret := os.Getenv("ARM_CLIENT_SECRET")
+	randomUUID, _ := uuid.GenerateUUID()
+	password := generateRandomPassword(10)
+	return d.basic(data, clientSecret, randomUUID, password)
+}
```

Comment:
For consistency the template should exist within the resource's tests and be referenced by the data source tests, not the other way around.

Reply:
Ok, Let me change this.